internal/services/aws_reservation_service.go

```diff
@@ -139,7 +163,27 @@ func CreateAWSReservation(w http.ResponseWriter, r *http.Request) {
 		logger.Debug().Msgf("Found existing pubkey resource %d, upload not enqueued", pkr.ID)
 	}
 
+	logger.Debug().Msgf("Enqueuing launch instance job for pubkey %d and source %d", pk.ID, reservation.SourceID)
+
+	argsLaunch := &jobs.LaunchInstanceAWSTaskArgs{
+		AccountID:     accountId,
+		ReservationID: reservation.ID,
+		PubkeyID:      pk.ID,
+		ImageID:       reservation.ImageID,
+		ARN:           arn,
+		Amount:        reservation.Amount,
+		InstanceType:  reservation.InstanceType,
+		//TODO: NEEDS THE RESOURCE NAME LIKE "lzap-rsa-2010-{{$randomInt}}"
+		// PubkeyHandle:  pkr.Name,
+	}
```

Comment:
Yeah see my patch https://github.com/RHEnVision/provisioning-backend/pull/135

You can simply pass Pubkey.Name here now, its not unique now but it will be once my PR is merged.

Reply:
This comment is outdated, I have deleted it.
But I am not sure we should store the pubkey name in addition to the pubkey id. 
Now I am fetching pubkey by id from the DB. 
What do you think is better?

Reply:
Yeah I thought about it too, then I deleted my comment to keep it simple. Since you started...

Both are doable, but I would probably keep just passing the ID and fetching it later, because if a customer will hit a problem that an already existing key has a name that cannot be used in our database, we will probably do some renaming routine that will name the key something like "my_key-2" and we would need to store such name in PubkeyResource. So fetching would be the only option anyway to support these renamed keys.